viz_scripts/scaffolding.py

```diff
@@ -126,50 +131,35 @@ def load_viz_notebook_data(year, month, program, study_type, has_dynamic_labels,
     # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
     if "distance" in expanded_ct.columns:
         unit_conversions(expanded_ct)
-
-    # Extract translations_data for dynamic_labels
-    if has_dynamic_labels:
-        print("This json has dynamic label")
-        req = requests.get(dynamic_labels_url)
-        if req.status_code != 200:
-            print("Unable to download dynamic_labels")
-        else:
-            print("dynamic labels download was successful.")
-            dynamic_labels = json.loads(req.text)
-
-        # Extract translations section
-        translations_data = dynamic_labels["translations"]["en"]
-
-        # Map new mode labels with translations_data from dynamic_labels
-        # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
-        if "mode_confirm" in expanded_ct.columns:
-            expanded_ct["Mode_confirm"] = expanded_ct["mode_confirm"].map(translations_data)
-        if study_type == 'program':
-            # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
-            if 'replaced_mode' in expanded_ct.columns:
-                expanded_ct['Replaced_mode']= expanded_ct['replaced_mode'].map(translations_data)
-            else:
-                print("This is a program, but no replaced modes found. Likely cold start case. Ignoring replaced mode mapping")
-        else:
-                print("This is a study, not expecting any replaced modes.")
+    
+    # Extract translations key
+    dic_translations = dict()
+    if "translations" in dynamic_labels and "en" in dynamic_labels["translations"]:
+        dic_translations = dynamic_labels["translations"]["en"]
+
+    # Select the mapping based on availability of dynamic_labels
+    if dic_translations:
+        dic_mapping = dic_translations
     else:
-        # Mapping new mode labels with dictionaries
+        dic_mapping = dic_re
+    
+    # Map new mode labels with translations dictionary from dynamic_labels
+    # CASE 2 of https://github.com/e-mission/em-public-dashboard/issues/69#issuecomment-1256835867
+    if "mode_confirm" in expanded_ct.columns:
+        update_expanded_ct(expanded_ct, "Mode_confirm", "mode_confirm", dic_mapping)
```

Comment:
`expanded_ct['Mode_confirm']= expanded_ct['mode_confirm'].map(dic_mapping)` seems like it would work just fine.

Reply:
Changed.